File: src/com/dafttech/terra/engine/lighting/Light.java
package com.dafttech.terra.engine.lighting;

import com.badlogic.gdx.graphics.Color;
import com.dafttech.terra.engine.AbstractScreen;
import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.entities.Entity;

public abstract class Light {
    Color color = Color.WHITE;
    
    public void setColor(Color c) {
        color = c;
    }
    
    public Color getColor() {
        return color;
    }
    
    public abstract void drawToLightmap(AbstractScreen screen, Entity pointOfView);

    public abstract void setPosition(Vector2 p);
}

File: src/main/java/com/dafttech/terra/engine/lighting/Light.scala
package com.dafttech.terra.engine.lighting

import com.badlogic.gdx.graphics.Color
import com.dafttech.terra.engine.{AbstractScreen, Vector2}
import com.dafttech.terra.game.world.entities.Entity

abstract class Light {
  private[lighting] var color: Color = Color.WHITE

  def setColor(c: Color) {
    color = c
    color.a = 1
  }

  def getColor: Color = {
    return color
  }

  def drawToLightmap(screen: AbstractScreen, pointOfView: Entity)

  def setPosition(p: Vector2)
}